oauth2/playground2/src/main/webapp/oauth2.jsp

```diff
@@ -687,10 +687,16 @@
                             <td><input type="submit" name="authorize" value="Get Access Token"></td>
                             <%
                                 if (isOIDCLogoutEnabled) {
+                                    String logout_url = (String)session.getAttribute(OAuth2Constants.OIDC_LOGOUT_ENDPOINT) + "?post_logout_redirect_uri=" + ApplicationConfig.getPostLogoutRedirectUri();
+                                    if (StringUtils.isNotBlank(idToken)) {
+                                        logout_url+= "&id_token_hint=" + idToken;
+                                    } else {
+                                        logout_url+= "&client_id=" + (String)session.getAttribute(OAuth2Constants.CONSUMER_KEY);
```

Comment:
If client id and idTokenHint both not available what will happen?

Reply:
If neither the client id nor the id token hint are available, only Identity Server logout will be possible, while federated IdP logout will not as it is dependent on these values to identify the relying party and the browser will be redirected to the default logout page.

Reply:
However, the client id is expected to be available in all the logout flows as this value is provided prior to login for all grant types.